src/components/Observations/UploadPrompt.js

```diff
@@ -5,29 +5,25 @@ import { t } from "i18next";
 import type { Node } from "react";
 import React from "react";
 import { Text } from "react-native";
+import useNumUnuploadedObservations from "sharedHooks/useNumUnuploadedObservations";
 
 type Props = {
-  numOfUnuploadedObs: number,
-  updateUploadStatus: Function,
-  uploadObservations: Function
+  startUpload: Function
 }
 
-const UploadPrompt = ( {
-  uploadObservations, numOfUnuploadedObs, updateUploadStatus
-}: Props ): Node => (
-  <>
-    <Text>{t( "Whenever-you-get-internet-connection-you-can-upload" )}</Text>
-    <Button
-      level="neutral"
-      text={t( "UPLOAD-X-OBSERVATIONS", { count: numOfUnuploadedObs } )}
-      className="py-1 mt-5"
-      onPress={( ) => {
-        updateUploadStatus( );
-        uploadObservations( );
-      }}
-    />
-
-  </>
-);
+const UploadPrompt = ( { startUpload }: Props ): Node => {
+  const numUnuploadedObs = useNumUnuploadedObservations( );
+  return (
+    <>
```

Comment:
Hmm, my understanding here is that it's actually better practice in tests to skip using testID and instead look for components by text (or role, etc.) where possible. That ensures that we're looking for the accessible elements in a test.

From the [info box](https://callstack.github.io/react-native-testing-library/docs/api-queries/#bytestid) in the testing docs:

`In the spirit of [the guiding principles](https://testing-library.com/docs/guiding-principles), it is recommended to use this only after the other queries don't work for your use case. Using testID attributes do not resemble how your software is used and should be avoided if possible. However, they are particularly useful for end-to-end testing on real devices, e.g. using Detox and it's an encouraged technique to use there.`

So question here: do we need the testID for Detox? Otherwise I'd prefer that we try to use best practices in our jest tests.

Reply:
Yes, for the unit tests or integration tests react-native-testing-library is strongly opinionated towards using accessible elements. E2e framework Detox is actually opinionated in the opposite way, to prefer testID. But we are striving to use user-facing information as well in the e2e tests.
However, the accessibility test is different in my view, it is more like an audit of the component (and traversing all the children) and for me testID is okay for that because I want to find the outermost container.

Reply:
Or alternatively, I'd be satisfied if you add the accessibility test for this component using the Fragment approach.

Reply:
Ok great, glad to understand where and when testID is important. I'm going to go ahead and change the fragment, rather than adding tests, since we're planning to get rid of this component in the redesigned UI.